hudi-common/src/main/java/org/apache/hudi/common/model/HoodieFileGroup.java

```diff
@@ -148,7 +172,10 @@ public Option<FileSlice> getLatestFileSlicesIncludingInflight() {
    */
   public Stream<FileSlice> getAllFileSlices() {
     if (!timeline.empty()) {
-      return fileSlices.values().stream().filter(this::isFileSliceCommitted);
+      List<String> instantsToRollback = getInstantsToRollbackFromTimeline();
+      return fileSlices.values().stream()
+          .filter(this::isFileSliceCommitted)
+          .filter(s -> !instantsToRollback.contains(s.getBaseInstantTime()));
```

Comment:
HoodieFileGroup is a core class and getAllFileSlices() is extensively used.  this is not the right place to be reading rollback metadata.  This will slow down almost all timeline operations.

Reply:
@bvaradar

Understood, am working on a hacky fix for this now... I'll try to avoid touching core APIs for now.